src/xclim/indices/_hydrology.py

```diff
@@ -685,3 +700,417 @@ def antecedent_precipitation_index(pr: xarray.DataArray, window: int = 7, p_exp:
+    ``annual_maxima`` and ``lag_snowpack_flow_peaks`` functions.
+
+    References
+    ----------
+    :cite:cts:`alonso_gonzalez_2022`
+    """
+    frz = convert_units_to(thresh, swe)
+    out = threshold_count(swe, ">", frz, freq)
+    return to_agg_units(out, swe, "count", deffreq="D")
+
+
+@declare_units(pr="[precipitation]", pet="[precipitation]")
+def aridity_index(pr: xarray.DataArray, pet: xarray.DataArray, freq: str = "YS") -> xarray.DataArray:
+    """
+    Aridity index.
+
+    The ratio of total precipitation over potential evapotranspiration.
+    Classification based on the Aridity Index (AI).
+    +----------------+----------------+-----------------+
+    | Classification | Aridity Index  | Global land area|
+    +----------------+----------------+-----------------+
+    | Hyperarid      | AI < 0.05      | 7.5%            |
+    | Arid           | 0.05 ≤ AI < 0.20 | 12.1%         |
+    | Semi-arid      | 0.20 ≤ AI < 0.50 | 17.7%         |
+    | Dry subhumid   | 0.50 ≤ AI < 0.65 | 9.9%          |
+    +----------------+----------------+-----------------+
+
+    Parameters
+    ----------
+    pr : array_like
+        Precipitation.
+    pet : array_like
+        Potential evapotranspiration.
+    freq : str, optional
+        Resampling frequency. A monthly or yearly frequency is expected.
+
+    Returns
+    -------
+    float
+        Aridity index per time step (Unitless).
+
+    Notes
+    -----
+    - An aridity index below 0.65 indicates an arid environment,
+      while values above this threshold correspond to more humid environments.
+    - In North America, higher aridity index values can be associated with colder climates
+      due to lower evapotranspiration, even when precipitation is limited or occurring as snow.
+
+    References
+    ----------
+    :cite:cts:'zomer_2022'
+    """
+    pr = pr.resample(time=freq).sum()
+    pet = pet.resample(time=freq).sum()
+    ai = pr / pet
+    ai.attrs["units"] = ""
+
+    return ai
+
+
+@declare_units(swe="[length]", q="[discharge]")
+def lag_snowpack_flow_peaks(
+    swe: xarray.DataArray,
+    q: xarray.DataArray,
+    freq: str = "YS-OCT",
+    percentile: int = 90,
+) -> xarray.DataArray:
+    """
+    Time lag between maximum snowpack and river high flows.
+
+    Number of days between the annual maximum snowpack, measured by the snow water
+    equivalent, and the mean date when river flow exceeds a percentile threshold
+    during a given year.
+    If the time lag between maximum snowpack and river high flows is ≤ 50 days,
+    the watershed is likely in a nival regime.
+
+    Parameters
+    ----------
+    swe : xarray.DataArray
+        Surface snow amount as snow water equivalent.
+    q : xarray.DataArray
+        Streamflow.
+    freq : str, optional
+        Resampling frequency. Defaults to the water year starting on the 1st of October.
+    percentile : float, optional
+        Percentile threshold identifying high flows. Defaults to the 90th percentile.
```

Comment:
These two variables cannot be passed to the function as `None`, so I would drop the `, optional`.